pkg/mnemonic/mnemonic_test.go

```diff
@@ -0,0 +1,107 @@
+package mnemonic
+
+import (
+	"strings"
+	"testing"
+	"unicode"
+)
+
+func TestPatterns(t *testing.T) {
+	for lengthKey, p := range patterns {
+		for i, pattern := range p {
+			if len(pattern) != lengthKey {
+				t.Errorf("len(patterns[%v][%v]) = %v; expected %v", lengthKey, i, len(pattern), lengthKey)
+			}
+
+			for _, partOfSpeech := range pattern {
+				if _, ok := validPartsOfSpeech[partOfSpeech]; !ok {
+					t.Errorf("patterns[%v][%v] contains invalid part of speech %q", lengthKey, i, partOfSpeech)
+				}
+			}
+		}
+	}
+}
+
+func TestVocabulary(t *testing.T) {
+	for partOfSpeech, _ := range validPartsOfSpeech {
+		if _, ok := vocabulary[partOfSpeech]; !ok {
+			t.Errorf("vocabulary has no entry for part of speech %q", partOfSpeech)
+			continue
+		}
+
+		for i := 0; i < 26; i++ {
+			letter := rune('a' + i)
+			if len(vocabulary[partOfSpeech]) <= i {
+				t.Errorf("vocabulary[%q] doesn't contain an entry for letter %q", partOfSpeech, letter)
+				break
+			}
+			word := vocabulary[partOfSpeech][i]
+			if len(word) == 0 || rune(word[0]) != letter {
+				t.Errorf("vocabulary[%q][%v] = %q; expected a word starting with %q", partOfSpeech, i, word, letter)
+				break
+			}
+		}
+	}
+}
+
+func isValidPhrase(phrase, acronym string) bool {
```

Comment:
Does this function correctly validate the number of words in a phrase?

Reply:
Good catch, I will add that test case.